### PR TITLE
Resolves a couple of open issues

### PR DESCRIPTION
**Pull request by mwpowellhtx**
Let me know if you have any concerns. Assuming this hasn't diverged too far from master/trunk, should be ready to go for merge.

* [Renamed the WIN files as MK files](http://github.com/google/or-tools/issues/331)
* [Handles known local SVN concerns](http://github.com/google/or-tools/issues/330)
* [Working around NMAKE/MAKE collisions for now](http://github.com/google/or-tools/issues/329)
* [Made a couple of attempts at resolving AssemblyInfo](http://github.com/google/or-tools/issues/323): methinks this is likely just the way it is, but am still interested to see the info from Explorer File Info
* Last but not least, culminating in [NuGet packaging/delivery](http://github.com/google/or-tools/issues/324): there was some confusion in the naming of the package, but I also took the opportunity to separate out some logical targets, as well as for the FlatZinc sub-project

### CODE REVIEW

File: makefiles/Makefile.win.mk [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+S = \\
 PROTOBUF_DIR = $(WINDOWS_PROTOBUF_DIR)
 SWIG_BINARY = $(WINDOWS_SWIG_BINARY)
 LIB_DIR = $(OR_ROOT)lib
 LIB_PREFIX =
 STATIC_LIB_SUFFIX = lib
 LIB_SUFFIX = lib
 BIN_DIR = $(OR_ROOT)bin
-GEN_DIR = $(OR_ROOT)src\\gen
+GEN_DIR = $(OR_ROOT)src$Sgen
```
Comment (lperron) [inline makefiles/Makefile.win.mk]: You know you are on windows. So \\ is fine IMO.

File: makefiles/Makefile.csharp.mk [hunk 2]
```diff
@@ -40,26 +51,26 @@ BASE_CLR_DLL_NAME:= $(CLR_DLL_NAME)
...
-	$(BIN_DIR)/cslinearprogramming$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/csls_api$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/csflow$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/csrabbitspheasants$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/cstsp$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/furniture_moving_intervals$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/organize_day_intervals$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/techtalk_scheduling$(CLR_EXE_SUFFIX).exe \
-	$(BIN_DIR)/cscvrptw$(CLR_EXE_SUFFIX).exe
+	$(BIN_DIR)/csknapsack$(CLR_EXE_SUFFIX)$(E) \
```
Comment (lperron) [inline makefiles/Makefile.csharp.mk]: You break unix build $E = "" on unix.

File: makefiles/Makefile.csharp.mk [hunk 3]
```diff
@@ -246,202 +250,235 @@ $(BIN_DIR)/$(CLR_DLL_NAME)$DLL: \
...
-
-$(BIN_DIR)/cstsp$(CLR_EXE_SUFFIX).exe: $(BIN_DIR)/$(CLR_DLL_NAME)$DLL $(EX_DIR)/csharp/cstsp.cs
-	$(CSC) $(SIGNING_FLAGS) /target:exe /out:$(BIN_DIR)$Scstsp$(CLR_EXE_SUFFIX).exe /platform:$(NETPLATFORM) /lib:$(BIN_DIR) /r:$(CLR_DLL_NAME)$DLL /r:Google.Protobuf$DLL $(EX_DIR)$Scsharp$Scstsp.cs
-
-$(BIN_DIR)/cscvrptw$(CLR_EXE_SUFFIX).exe: $(BIN_DIR)/$(CLR_DLL_NAME)$DLL $(EX_DIR)/csharp/cscvrptw.cs
-	$(CSC) $(SIGNING_FLAGS) /target:exe /out:$(BIN_DIR)$Scscvrptw$(CLR_EXE_SUFFIX).exe /platform:$(NETPLATFORM) /lib:$(BIN_DIR) /r:$(CLR_DLL_NAME)$DLL /r:Google.Protobuf$DLL $(EX_DIR)$Scsharp$Scscvrptw.cs
-
-$(BIN_DIR)/csls_api$(CLR_EXE_SUFFIX).exe: $(BIN_DIR)/$(CLR_DLL_NAME)$DLL $(EX_DIR)/csharp/csls_api.cs
-	$(CSC) $(SIGNING_FLAGS) /target:exe /out:$(BIN_DIR)$Scsls_api$(CLR_EXE_SUFFIX).exe /platform:$(NETPLATFORM) /lib:$(BIN_DIR) /r:$(CLR_DLL_NAME)$DLL /r:Google.Protobuf$DLL $(EX_DIR)$Scsharp$Scsls_api.cs
+$(BIN_DIR)/csrabbitspheasants$(CLR_EXE_SUFFIX)$(E): \
```
Comment (lperron) [inline makefiles/Makefile.csharp.mk]: same comment on unix.

**Comment by mwpowellhtx** [comment]
@lperron My mistake. I will submit a correction ASAP. Thanks!

On March 7, 2017 5:24:29 AM EST, lperron <notifications@github.com> wrote:
>lperron commented on this pull request.
>
>
>
>>      endif
>   endif
> endif
> 
> CSHARPEXE = \
>-	$(BIN_DIR)/csknapsack$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/csintegerprogramming$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/cslinearprogramming$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/csls_api$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/csflow$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/csrabbitspheasants$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/cstsp$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/furniture_moving_intervals$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/organize_day_intervals$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/techtalk_scheduling$(CLR_EXE_SUFFIX).exe \
>-	$(BIN_DIR)/cscvrptw$(CLR_EXE_SUFFIX).exe
>+	$(BIN_DIR)/csknapsack$(CLR_EXE_SUFFIX)$(E) \
>
>You break unix build $E = "" on unix.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/google/or-tools/pull/333#pullrequestreview-25477530

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by mwpowellhtx** [comment]
@lperron Okay, what is the purpose of the ``CSHARPEXE`` variable? It seems to be there only to serve the ``csharpexe`` target?

```
CSHARPEXE = \
	$(BIN_DIR)/csknapsack$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/csintegerprogramming$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/cslinearprogramming$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/csls_api$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/csflow$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/csrabbitspheasants$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/cstsp$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/furniture_moving_intervals$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/organize_day_intervals$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/techtalk_scheduling$(CLR_EXE_SUFFIX).exe \
	$(BIN_DIR)/cscvrptw$(CLR_EXE_SUFFIX).exe
```

I do see that ``csknapsack.exe`` landed in the bin/ folder. Are you saying that *.exe* should happen for ``*nix`` as well?

**Comment by mwpowellhtx** [comment]
Renaming these was intentional. These are critical for proper comparison:

```
makefiles/Makefile.third_party.unix
makefiles/Makefile.third_party.win
```

Otherwise, diff tools thinks they are images, not text.

**Comment by mwpowellhtx** [comment]
@lperron Additionally, so FlatZinc is not ready for publishing, either? It's an interesting approach.

**Comment by mwpowellhtx** [comment]
Closing this one. Will re-submit.

**Comment by mwpowellhtx** [comment]
@lperron There is also value in separating the nuget targets, for OrTools, and for FlatZinc: 1) staging, 2) packing, and 3) distribution. This is because, for folks like me, we like to sometimes stage our package to local repositories for test purposes. And with the make target chaining, it is trivial to reconstruct a target in your CI.

**Pull request closed without merging**